packages/math/src/pool/concentrated/one-for-zero.ts

```diff
@@ -0,0 +1,111 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+
+import { maxSqrtRatio } from "./const";
+import { SwapStrategy } from "./swap-strategy";
+import {
+  calcAmount0Delta,
+  calcAmount1Delta,
+  getAmountRemainingLessFee,
+  getFeeChargePerSwapStepOutGivenIn,
+  getNextSqrtPriceFromAmount1RoundDown,
+} from "./utils";
+
+export type OneForZero = {
+  isOutGivenIn: boolean;
+  sqrtPriceLimit: Dec;
+  swapFee: Dec;
+};
+
+export class OneForZeroStrategy implements SwapStrategy {
+  constructor(private readonly oneForZero: OneForZero) {}
+
+  computeSwapStep(
+    curSqrtPrice: Dec,
+    targetSqrtPrice: Dec,
+    liquidity: Dec,
+    amountRemaining: Dec
+  ): {
+    nextSqrtPrice: Dec;
+    amountRemaining: Dec;
+    amountComputed: Dec;
+    feeChargeTotal: Dec;
+  } {
+    const sqrtPriceTarget = this.getSqrtTargetPrice(targetSqrtPrice);
+
+    let amountOne = calcAmount1Delta(liquidity, sqrtPriceTarget, curSqrtPrice);
+
+    const amountRemainingLessFee = getAmountRemainingLessFee(
+      amountRemaining,
+      this.oneForZero.swapFee,
+      this.oneForZero.isOutGivenIn
+    );
+
+    let nextSqrtPrice: Dec;
+    let hasReachedTarget = false;
+
+    if (amountRemainingLessFee.gte(amountOne)) {
+      nextSqrtPrice = sqrtPriceTarget;
+      hasReachedTarget = true;
```

Comment:
I don't think we should be setting this to true here. While unlikely, it is possible that on the following line:
```
 nextSqrtPrice = getNextSqrtPriceFromAmount1RoundDown(
        liquidity,
        curSqrtPrice,
        amountRemainingLessFee
      );
``` 

`nextSqrtPrice` will reach target. Therefore, I suggest doing a comparison check similar to: https://github.com/osmosis-labs/osmosis/blob/6014a7c6f7f330869469f6ffc15f62cff3b854c1/x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go#L65

Reply:
I changed this since it seemed to be a reference equality check? Not a value equality check?